packages/terra-table/src/subcomponents/Row.jsx

```diff
@@ -180,6 +181,8 @@ function Row(props) {
       {cells.map((cellData, index) => {
         const cellColumnIndex = index + columnIndexOffSet;
         const columnId = displayedColumns[cellColumnIndex].id;
+        const { columnSpan, columnSpanIndex } = displayedColumns[cellColumnIndex];
```

Comment:
The Row component should not care about the column span.

Reply:
Updated.